src/esp/sim/AbstractReplayRenderer.cpp

```diff
@@ -91,12 +91,23 @@ void AbstractReplayRenderer::setSensorTransformsFromKeyframe(
 }
 
 void AbstractReplayRenderer::render(
-    Cr::Containers::ArrayView<const Mn::MutableImageView2D> imageViews) {
-  CORRADE_ASSERT(imageViews.size() == doEnvironmentCount(),
-                 "ReplayRenderer::render(): expected" << doEnvironmentCount()
-                                                      << "image views but got"
-                                                      << imageViews.size(), );
-  return doRender(imageViews);
+    Corrade::Containers::ArrayView<const Magnum::MutableImageView2D>
+        colorImageViews,
+    Corrade::Containers::ArrayView<const Magnum::MutableImageView2D>
+        depthImageViews) {
+  if (colorImageViews != nullptr) {
+    CORRADE_ASSERT(colorImageViews.size() == doEnvironmentCount(),
+                   "ReplayRenderer::render(): expected"
+                       << doEnvironmentCount() << "color image views but got"
+                       << colorImageViews.size(), );
+  }
+  if (depthImageViews != nullptr) {
+    CORRADE_ASSERT(depthImageViews.size() == doEnvironmentCount(),
+                   "ReplayRenderer::render(): expected"
+                       << doEnvironmentCount() << "depth image views but got"
+                       << depthImageViews.size(), );
```

Comment:
Might want to use ESP_CHECK in both instead, as this is possible to be triggered from Python, and ESP_CHECK will do a proper Python exception instead of an abort.

Hmm, and maybe also check for formats / sizes? Not sure how far do we want to go with the checks tho :)